### PR TITLE
adds singleDateTime  handling to cmr-util getGranuleTemporalInfo

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:**

Addresses [CUMULUS-2836: Add SingleDateTime to CMR temporal code](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2836)

## Changes
*  Updates `cmr-utils/getGranuleTemporalInfo` to search for a single date time
    element, when a beginningDateTime and endingDateTime value are not found in
    the CMR metadata file.  The granule's temporal information is updated so
    that both beginningDateTime and endingDateTime are set to the singleDateTimeValue.
*  adds tests and fixtures for each valid metadata file type (cmr json, cmr xml and iso xml)
* [echo-10 metadata schema](https://git.earthdata.nasa.gov/projects/EMFD/repos/echo-schemas/browse/schemas/10.0/Granule.xsd#1620) 
* [UMM-G doc](https://wiki.earthdata.nasa.gov/download/attachments/49448405/UMM-G_V1.6.3_20210503.docx?version=1&modificationDate=1629836881407&api=v2), for translations p38-41

Additional PR #2670 to reformat the code after the initial review.

## PR Checklist

- [ x] Update CHANGELOG
- [ x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/cmrjs/tests/fixtures/ATL03_fixture.SingleDateTime.cmr_iso.xml [hunk 1]
```diff
@@ -0,0 +1,1985 @@
...
+                        <gml:TimeInstant gml:id="TIME_PERIOD_ID">
+                          <gml:timePosition>2022-01-18T00:04:18.000Z</gml:timePosition>
+                        </gml:TimeInstant>
+                      </gmd:extent>
+                    </gmd:EX_TemporalExtent>
+                  </gmd:temporalElement>
+                </gmd:EX_Extent>
+              </gmd:extent>
+            </gmd:MD_DataIdentification>
+          </gmd:identificationInfo>
```
Comment (flamingbear) [inline packages/cmrjs/tests/fixtures/ATL03_fixture.SingleDateTime.cmr_iso.xml]: Changed from original fixture

File: packages/cmrjs/tests/fixtures/MOD09GQ.singleDateTime.cmr.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      {
+        "Name": "Not provided",
+        "Size": 1.009857177734375,
+        "SizeUnit": "NA"
+      }
+    ]
+  },
+  "TemporalExtent": {
+    "SingleDateTime": "2022-01-18T14:40:00.000Z"
+  },
```
Comment (flamingbear) [inline packages/cmrjs/tests/fixtures/MOD09GQ.singleDateTime.cmr.json:43]: Updated from original fixture

File: packages/cmrjs/tests/fixtures/cmrFileUpdateFixture.SingleDateTime.cmr.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    <ReprocessingPlanned>The Reprocessing Planned Statement Value</ReprocessingPlanned>
+    <ReprocessingActual>The Reprocessing Actual Statement Value</ReprocessingActual>
+    <ProducerGranuleId>SMAP_L3_SM_P_20150407_R13080_001.h5</ProducerGranuleId>
+    <DayNightFlag>UNSPECIFIED</DayNightFlag>
+    <ProductionDateTime>2018-07-19T12:01:01Z</ProductionDateTime>
+    <LocalVersionId>LocalVersionIdValue</LocalVersionId>
+  </DataGranule>
+  <Temporal>
+    <SingleDateTime>2022-01-24T00:00:00Z</SingleDateTime>
+  </Temporal>
```
Comment (flamingbear) [inline packages/cmrjs/tests/fixtures/cmrFileUpdateFixture.SingleDateTime.cmr.xml:21]: updated from original fixture

File: packages/cmrjs/src/cmr-utils.js [hunk 4]
```diff
@@ -1091,28 +1092,39 @@ async function getGranuleTemporalInfo(granule) {
...
     const identificationInfo = metadataMI['gmd:identificationInfo'];
     const dataIdentification = identificationInfo.find((dataIdObject) => Object.keys(dataIdObject).filter((key) => Object.keys(dataIdObject[key]).includes('gmd:extent')));
-    const temporalInfo = dataIdentification['gmd:MD_DataIdentification']['gmd:extent']['gmd:EX_Extent']['gmd:temporalElement']['gmd:EX_TemporalExtent']['gmd:extent']['gml:TimePeriod'];
-    const beginningDateTime = temporalInfo['gml:beginPosition'];
-    const endingDateTime = temporalInfo['gml:endPosition'];
+    const temporalInfo = get(dataIdentification, 'gmd:MD_DataIdentification.gmd:extent.gmd:EX_Extent.gmd:temporalElement.gmd:EX_TemporalExtent.gmd:extent.gml:TimePeriod');
+    let beginningDateTime = get(temporalInfo, 'gml:beginPosition');
+    let endingDateTime = get(temporalInfo, 'gml:endPosition');
+
+    if (!beginningDateTime && !endingDateTime) {
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply packages/cmrjs/src/cmr-utils.js:1104]: agree we should be consistent probably

File: packages/cmrjs/src/cmr-utils.js [hunk 5]
```diff
@@ -1091,28 +1092,39 @@ async function getGranuleTemporalInfo(granule) {
...
+    const metadataMI = get(metadata, 'gmd:DS_Series.gmd:composedOf.gmd:DS_DataSet.gmd:has.gmi:MI_Metadata');
 
     // Get beginning and ending date time from beginPosition and endPosition
     const identificationInfo = metadataMI['gmd:identificationInfo'];
     const dataIdentification = identificationInfo.find((dataIdObject) => Object.keys(dataIdObject).filter((key) => Object.keys(dataIdObject[key]).includes('gmd:extent')));
-    const temporalInfo = dataIdentification['gmd:MD_DataIdentification']['gmd:extent']['gmd:EX_Extent']['gmd:temporalElement']['gmd:EX_TemporalExtent']['gmd:extent']['gml:TimePeriod'];
-    const beginningDateTime = temporalInfo['gml:beginPosition'];
-    const endingDateTime = temporalInfo['gml:endPosition'];
+    const temporalInfo = get(dataIdentification, 'gmd:MD_DataIdentification.gmd:extent.gmd:EX_Extent.gmd:temporalElement.gmd:EX_TemporalExtent.gmd:extent.gml:TimePeriod');
+    let beginningDateTime = get(temporalInfo, 'gml:beginPosition');
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/cmrjs/src/cmr-utils.js:1101]: CMR sample xml file only uses beginPosition and endPosition

**Review by jennyhliu — APPROVED** [review]

**Pull request merged**